### PR TITLE
WEBDEV-5981 Make date picker full-width in mobile layout

### PR DESCRIPTION
**Pull request by latonv**
At mobile widths, the date picker only takes up a fraction of the available space, making it small and hard to read.

While in desktop view we solved this by allowing it to expand into a larger modal, in the mobile layout it is possible to make the date picker wider all the time, no modal required. This PR allows it to fill the entire container width available to it in mobile view.

This also entails upgrading `histogram-date-range` component version, as the previous version has a bug causing it not to fully render updates to the component's `width`/`height` properties.

### CODE REVIEW

File: src/collection-browser.ts [hunk 1]
```diff
@@ -384,44 +386,75 @@ export class CollectionBrowser
   }
 
   private get collectionBrowserTemplate() {
```
Comment (latonv) [inline src/collection-browser.ts:388]: The old version of this template was getting a bit bloated and hard to read, so I've split it out into several smaller sub-templates that should hopefully be easier to understand.

File: src/collection-browser.ts [hunk 2]
```diff
@@ -869,7 +903,8 @@ export class CollectionBrowser
   handleResize(entry: ResizeObserverEntry): void {
     const previousView = this.mobileView;
     if (entry.target === this.contentContainer) {
-      this.mobileView = entry.contentRect.width < this.mobileBreakpoint;
+      this.contentWidth = entry.contentRect.width;
+      this.mobileView = this.contentWidth < this.mobileBreakpoint;
```
Comment (latonv) [inline src/collection-browser.ts:907]: Preserving the contentRect's width to be passed down into the date picker (which requires an explicit width to render its svg).

File: src/collection-facets.ts [hunk 3]
```diff
@@ -229,10 +231,6 @@ export class CollectionFacets extends LitElement {
       : nothing;
   }
 
-  private get currentYearsHistogramAggregation(): Aggregation | undefined {
-    return this.aggregations?.year_histogram;
-  }
-
```
Comment (latonv) [inline src/collection-facets.ts:235]: (Obsolete, unused method)

**Review by iisa — APPROVED** [review]
looks good! thanks for the walk thru and splitting up code for good practice ^_^

**Pull request merged**